bonfire/openshift.py

```diff
@@ -85,21 +85,35 @@ def _resources_for_ns_wait():
 def _all_resources_ready(namespace, timeout, watcher):
     already_waited_on = set()
 
-    # wait on ClowdEnvironment, if there's one using this ns as its targetNamespace
+    # wait on ClowdEnvironments if any ClowdApps reference one
     start = time.time()
 
-    clowd_env = find_clowd_env_for_ns(namespace)
-    if clowd_env:
+    clowd_envs = []
+
+    clowdapps = get_json("clowdapp", namespace=namespace)
```

Comment:
You could simplify this a bit if clowd_envs was a set because then you can just use add and it wont add duplicates. I wouldn't die on this hill, but wanted to throw it out there. I also made getting stuff off the clowdapps json a bit safer, I don't know if there's any danger there

```
clowd_envs = set()

clowdapps = get_json("clowdapp", namespace=namespace)
for clowdapp in clowdapps.get("items", []):
    env_name = clowdapp.get("spec", {}).get("envName")
    if env_name:
        log.info("will wait on ClowdEnvironment '%s' found on ClowdApp .spec.envName", env_name)
        clowd_envs.add(env_name)
```

Reply:
You're right, I had it as a 'set' at first. But then, I want to check if it is already in the set in order to only log the message once and I went back to a list with it. I can make it a set, no problem.